channels/telegram/src/main/kotlin/com/justai/jaicf/channel/telegram/TelegramReactions.kt

```diff
@@ -18,11 +19,34 @@ class TelegramReactions(
 ) : Reactions() {
 
     val chatId = request.chatId
+    private val messages = mutableListOf<Message>()
+
+    private fun addResponse(pair: Pair<retrofit2.Response<Response<Message>?>?, Exception?>) {
+        pair.first?.body()?.result?.let { messages.add(it) }
+    }
 
     override fun say(text: String): SayReaction {
         return sendMessage(text)
     }
 
+    override fun buttons(vararg buttons: String): ButtonsReaction {
+        messages.lastOrNull()?.let { message ->
+            val keyboard = message.replyMarkup?.inlineKeyboard?.toMutableList() ?: mutableListOf()
+            keyboard.addAll(buttons.map { listOf(InlineKeyboardButton(it, callbackData = it)) })
+
+            api.editMessageReplyMarkup(
+                    chatId,
+                    message.messageId,
+                    replyMarkup = InlineKeyboardMarkup(keyboard)
+            ).also {
+                messages.remove(message)
+                addResponse(it)
+            }
+        }
+
+        return ButtonsReaction.create(buttons.asList())
```

Comment:
If there is no previous response (i.e. `messages.lastOrNull == null`) and we are unable to send buttons, we anyway will log the `ButtonReaction`. Is this the desired behavior?

Reply:
Like for any other channel we cannot be sure that buttons was rendered successfully. And nevertheless we register buttons response as a content that was generated by scenario.

Reply:
Default implementations of `Reactions` methods don't register reactions, they just create them. So if a channel doesn't support some kind of response (and so doesn't override the method), the reaction will not be registered. So I think this is the case.

Reply:
`Reacion` has an internal constructor and could be created with `create` factory only. My point was about other channels - we cannot guarantee that some components (like buttons) will be rendered actually but nevertheless register the output..

Reply:
Ah, yes, you are right about a constructor, I missed it. 
But, anyway, when we can't guarantee that responses will not be rendered on a 3rd party side - it's okay to register them in order to tell a user that we've done all required work on the framework side.
But in the current case, the channel knows for sure that the reaction will not be even processed somehow, so in this case, the channel should not register the reaction, otherwise, it will lead to confusion for the user.
I'd suggest creating an issue for that after merge.